torchao/quantization/quant_primitives.py

```diff
@@ -2002,7 +2002,12 @@ def choose_qparams_affine_float8(
         ]
         scale = scale.reshape(output_shape)
 
-    return scale.to(dtype=scale_dtype)
+    if scale_dtype is not torch.float32:
+        # Shielding for Version > 2.8
+        assert scale_dtype is torch.float8_e8m0fnu, "Only float8_e8m0fnuz is supported"
+        scale = torch.exp2(torch.round(torch.log2(scale)))
+
+    return scale.to(dtype=torch.float32)
```

Comment:
IMO this is a really great way to express this, this switch back is the only spooky part